kratos/containers/model.cpp

```diff
@@ -27,108 +27,146 @@
-                
+
         }
-        
+
         //add the root model part to the list
-        ModelPart& root_model_part = pmodel_part->GetRootModelPart();
-        mroot_map[root_model_part.Name()] = &root_model_part;
-        
+        ModelPart& root_model_part = pModelPart->GetRootModelPart();
+        mRootModelPartMap[root_model_part.Name()] = &root_model_part;
+
         KRATOS_CATCH("")
     }
 
-    void Model::AddModelPartRawPointer( ModelPart* pmodel_part) //TODO: DEPRECATED. to be removed. this is TEMPORARY
+    void Model::AddModelPartRawPointer( ModelPart* pModelPart) //TODO: DEPRECATED. to be removed. this is TEMPORARY
     {
         KRATOS_TRY
-        
+
         //TODO: flat map should disappear in the future!!
-        auto search = mflat_map.find(pmodel_part->Name());
+        auto search = mflat_map.find(pModelPart->Name());
         if( search == mflat_map.end())
         {
-            mflat_map[pmodel_part->Name()] = pmodel_part;
-            
+            mflat_map[pModelPart->Name()] = pModelPart;
+
             //walk the submodelparts
-            for(auto& part : pmodel_part->SubModelParts())
+            for(auto& part : pModelPart->SubModelParts())
                 AddModelPartRawPointer(&part);
         }
         else
         {
-            if(&(*search->second) != &*(pmodel_part))
+            if(&(*search->second) != &*(pModelPart))
                 KRATOS_ERROR << "trying to add to the Model two DISTINCT model parts with the same name. This should be possible (and it will be in the future) if they belong to two different root model_parts, but it is currently disallowed";
-                
+
         }
-        
+
         //add the root model part to the list
-        ModelPart& root_model_part = pmodel_part->GetRootModelPart();
-        mroot_map[root_model_part.Name()] = &root_model_part;
-        
+        ModelPart& root_model_part = pModelPart->GetRootModelPart();
+        mRootModelPartMap[root_model_part.Name()] = &root_model_part;
+
         KRATOS_CATCH("")
     }
-    
-    ModelPart& Model::GetModelPart(std::string name)
+
+    ModelPart& Model::GetModelPart(const std::string& rFullModelPartName)
     {
         KRATOS_TRY
 
-        KRATOS_ERROR_IF( name.empty() ) << "Attempting to find a model part with empty name (\"\")!" << std::endl;
-                
-        auto search = mflat_map.find(name);
+        KRATOS_ERROR_IF( rFullModelPartName.empty() ) << "Attempting to find a "
+            << "ModelPart with empty name (\"\")!" << std::endl;
+
+        auto search = mflat_map.find(rFullModelPartName);
         if(search != mflat_map.end()) {
-            //TODO: issue a warning message telling that this behaviour is DEPRECATED and that it will be removed
+            KRATOS_WARNING_IF("Model", (search->second)->IsSubModelPart())
```

Comment:
@philbucher this feature is used pervasively, and the gui is not yet ready to get rid of this. so for now let's wait in activating this warning